core/src/main/java/jenkins/util/groovy/GroovyHookScript.java

```diff
@@ -40,9 +39,15 @@
 public class GroovyHookScript {
     private final String hook;
     private final Binding bindings = new Binding();
+    private final ServletContext servletContext;
+    private final File home;
+    private final ClassLoader loader;
 
-    public GroovyHookScript(String hook) {
+    public GroovyHookScript(String hook, @Nonnull ServletContext servletContext, @Nonnull File home, @Nonnull ClassLoader loader) {
```

Comment:
:bug: breaks the binary compatibility. This constructor is being used outside (search in CB repo)